app/models/resource_action_workflow.rb

```diff
@@ -44,9 +44,9 @@ def process_request(state)
     else
       ra = load_resource_action(values)
       if ra.resource.try(:open_url?)
-        result[:task_id] = ra.deliver_to_automate_from_dialog_with_miq_task(values, @target, @requester)
+        result[:task_id] = ra.deliver_task(values, @target, @requester)
```

Comment:
TODO is there a better way to check if we should create a task than checking a for just `open_url?`?  ref: [`CustomButton#open_url?`](https://github.com/ManageIQ/manageiq/blob/master/app/models/custom_button.rb#L246-L248)

Reply:
That is strange.
For core, it only lives in `CustomButton` and these 2 spots are the only place it is mentioned.
For ui-classic There is some mention around buttons and dialogs, but it seems minimal.

So open_url means it is an external resource so it should be done with a task? (rather than inline?)
I remember we used to do automate inline (in rails and not the worker) and had to introduce a mechanism to allow us to use a worker. Do you think this is related to that?